docs/source/traits_api_reference/trait_types.rst

```diff
@@ -11,6 +11,9 @@ Traits
   The ``Long``, ``CLong``, ``BaseLong`` and ``BaseCLong`` trait types are deprecated,
   and are now direct aliases for the corresponding ``Int`` trait types.
 
+.. deprecated:: 6.0
+  The ``DictStrLong`` trait type is deprecated.
```

Comment:
if we are only deprecating it then this comment should mention the fact that the `DictStrLong` is an alias for the `DictStrInt` (im not sure if this comment has been completely removed in a later commit)